tequilapi/http_middlewares_test.go

```diff
@@ -22,6 +22,25 @@ func TestCorsHeadersAreAppliedToResponse(t *testing.T) {
 	assert.True(t, mock.wasCalled)
 }
 
+func TestPreflightCorsCheckIsHandled(t *testing.T) {
+	req, err := http.NewRequest(http.MethodOptions, "/not-important", nil)
+	assert.NoError(t, err)
+
+	req.Header.Add("Access-Control-Request-Method", "DELETE")
+	req.Header.Add("Access-Control-Request-Headers", "origin, x-requested-with")
```

Comment:
Can we add content-type header here, to check that it's allowed? (which was why it was failing before).

Reply:
Refactored handler now simply allows all headers defined in request